### ISSUE TITLE
systemd --user accepts FailureAction then fails to execute it

### ISSUE THREAD

**Issue by keszybz** [systemd/systemd#1935]
Most likely FailureAction should be refused for --user units, since none of the options (reboot, poweroff) make sense for user instances.

I hit this when testing dnf-system-upgrade.service in user mode:

```
Nov 17 12:56:36 bupkis systemd[1952]: Starting System Upgrade...
Nov 17 12:56:36 bupkis systemd[1952]: Failed to load environment files: No such file or directory
Nov 17 12:56:36 bupkis systemd[1952]: dnf-system-upgrade.service failed to run 'start' task: No such file or directory
Nov 17 12:56:36 bupkis systemd[1952]: dnf-system-upgrade.service changed dead -> failed
Nov 17 12:56:36 bupkis systemd[1952]: Job dnf-system-upgrade.service/start finished, result=failed
Nov 17 12:56:36 bupkis systemd[1952]: Failed to start System Upgrade.
Nov 17 12:56:36 bupkis systemd[1952]: Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/systemd1 interface=org.freedesktop.systemd1.Manager member=JobRemoved cookie=4 reply_cookie=0 error=n/a
Nov 17 12:56:36 bupkis systemd[1952]: Unit dnf-system-upgrade.service entered failed state.
Nov 17 12:56:36 bupkis systemd[1952]: dnf-system-upgrade.service failed, rebooting.
Nov 17 12:56:36 bupkis systemd[1952]: Failed to load configuration for reboot.target: No such file or directory
Nov 17 12:56:36 bupkis systemd[1952]: Trying to enqueue job reboot.target/start/replace
Nov 17 12:56:36 bupkis systemd[1952]: Failed to reboot: Unit reboot.target failed to load: No such file or directory..
Nov 17 12:56:36 bupkis systemd[1952]: Collecting reboot.target
```

**Comment by andrewshadura** [comment]
Isn't systemd currently ignoring any actions in the user mode? There some code in [failure-action.c](../blob/53f18416/src/core/failure-action.c#L52) that seems to do that.

However, it may be useful to let systemd in the user to reboot the system if the user has enough rights to do so. Actually, we've run into exactly this: we have a service running in a user session which is critical and the system must be rebooted if it fails.

**Comment by poettering** [comment]
Fixed since 54fcb6192c618726d11404b24b1a1e9ec3169ee1
